informer/src/handler/rpc_ws_handler.rs

```diff
@@ -0,0 +1,94 @@
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{
+    start_ws, Connection, Error, ErrorCode, Params, PubSubHandler, Rng, Session, Subscriber, SubscriptionId, Value,
+    WsError, WsServer,
+};
+use crossbeam::Sender;
+use crossbeam_channel as crossbeam;
+use jsonrpc_core::{futures, BoxFuture};
+use std::io;
+use std::sync::Arc;
+
+pub struct InformerConfig {
+    pub interface: String,
+    pub port: u16,
+    pub max_connections: usize,
+}
+
+pub struct Handler {
+    pub handler: PubSubHandler<Arc<Session>>,
+}
+
+impl Handler {
+    pub fn new(handler: PubSubHandler<Arc<Session>>) -> Self {
+        Self {
+            handler,
+        }
+    }
+    pub fn start_ws(self, config: InformerConfig) -> Result<WsServer, String> {
+        let url = format!("{}:{}", config.interface, config.port);
+        let addr = url.parse().map_err(|_| format!("Invalid WebSockets listen host/port given: {}", url))?;
+        let server = self.handler;
+        let start_result = start_ws(&addr, server, config.max_connections);
+        match start_result {
+            Err(WsError::Io(ref err)) if err.kind() == io::ErrorKind::AddrInUse => {
+                Err(format!("WebSockets address {} is already in use, make sure that another instance of a Codechain node is not using this", addr))
+            },
+            Err(e) => {
+                Err(format!("WebSockets error: {:?}", e))
+            },
+            Ok(server) => {
+                cinfo!(INFORMER, "WebSockets Listening on {}", addr);
+                Ok(server)
+            },
+        }
+    }
+    pub fn event_subscription(&mut self, sender: Sender<Connection>) {
+        self.handler.add_subscription(
+            "register",
+            ("register", move |params: Params, _, subscriber: Subscriber| {
+                if params == Params::None {
+                    subscriber
+                        .reject(Error {
+                            code: ErrorCode::ParseError,
+                            message: "Invalid parameters. Subscription rejected.".into(),
+                            data: None,
+                        })
+                        .expect("Invalid parameters are used. Subscription rejected");
+                    return
+                }
+                let unwrapped_params: Result<Vec<String>, Error> = params.parse();
+                let all_params =
+                    unwrapped_params.expect("The parameter that the client sent for subscription is not valid");
+                let mut rng = rand::thread_rng();
+                let sub_id = rng.gen();
+                let sink = subscriber
+                    .assign_id(SubscriptionId::Number(sub_id))
+                    .expect("Subscription request is already terminated");
+                let mut connection = Connection::new(true, sink, sub_id);
```

Comment:
It's hard to know what it is when you just pass the `true` value.
I have two suggestions.
* Use an enum for it like "ConnectionState::Connected" and "ConnectionState::Disconnected".
* Do not receive the state from the `new` function. You will do not want to create a connection with the state value `false`.